c10/core/DispatchKeySet.h

```diff
@@ -195,102 +467,130 @@ class DispatchKeySet final {
+        DispatchKey::CustomRNGKeyId,
+        DispatchKey::MkldnnCPU,
+        DispatchKey::Meta,
+        // Sparse and Quantized backends also live here.
+        DispatchKey::Sparse,
+        DispatchKey::Quantized})
+        | DispatchKeySet(DispatchKeySet::RAW, full_backend_mask);
 
 // The set of dispatch keys that come after autograd
 // n.b. this relies on the fact that AutogradOther is currently the lowest
 // Autograd key
 constexpr DispatchKeySet after_autograd_keyset =
-    DispatchKeySet(DispatchKeySet::FULL_AFTER, c10::DispatchKey::AutogradOther);
+        DispatchKeySet(DispatchKeySet::FULL_AFTER, c10::DispatchKey::AutogradOther);
 
 // The set of dispatch keys that come after ADInplaceOrView
 constexpr DispatchKeySet after_ADInplaceOrView_keyset = DispatchKeySet(
-    DispatchKeySet::FULL_AFTER,
-    c10::DispatchKey::ADInplaceOrView);
+        DispatchKeySet::FULL_AFTER,
+        c10::DispatchKey::ADInplaceOrView);
 
 // The set of dispatch keys that come after Functionalize
 constexpr DispatchKeySet after_func_keyset =
-    DispatchKeySet(DispatchKeySet::FULL_AFTER, c10::DispatchKey::Functionalize)
-        .remove(
-            // NOTE: we also need to remove ADInplaceOrView from the keyset when
-            // redispatching after the func kernels. This is because we're not
-            // calling the same op; we originally called an inplace op, and now
-            // we aren't. The original key calculation figured out which keys
-            // were Fallthrough based on the inplace op. That means that it did
-            // not include the ADInPlaceOrView kernel as a fallthrough key.
-            // However, we WANT the ADInPlaceOrView kernel to be ignored now
-            // that we're calling an out-of-place op. Re-invoking
-            // Dispatcher::call would re-run the Fallthrough key calculation and
-            // get us that, But at::redispatch is more performant. We can get
-            // away with it by explicitly removing the key here.
-            c10::DispatchKey::ADInplaceOrView);
+        DispatchKeySet(DispatchKeySet::FULL_AFTER, c10::DispatchKey::Functionalize)
+            .remove(
+                // NOTE: we also need to remove ADInplaceOrView from the keyset when
+                // redispatching after the func kernels. This is because we're not
+                // calling the same op; we originally called an inplace op, and now
+                // we aren't. The original key calculation figured out which keys
+                // were Fallthrough based on the inplace op. That means that it did
+                // not include the ADInPlaceOrView kernel as a fallthrough key.
+                // However, we WANT the ADInPlaceOrView kernel to be ignored now
+                // that we're calling an out-of-place op. Re-invoking
+                // Dispatcher::call would re-run the Fallthrough key calculation and
+                // get us that, But at::redispatch is more performant. We can get
```

Comment:
Not in this PR but this will do the wrong thing for any op where the inplace and out of place versions have different fallthrough.

Reply:
Good catch 😄